crates/router/src/routes/app.rs

```diff
@@ -484,6 +493,7 @@ impl AppState {
             opensearch_client: Arc::clone(&self.opensearch_client),
             grpc_client: Arc::clone(&self.grpc_client),
             theme_storage_client: self.theme_storage_client.clone(),
+            request_headers: None,
```

Comment:
We may be able to pass `HeaderMap` to `get_session_state()` itself and have the field populated here. Then we need not have it as an `Option` either.

@jarnura What do you think, should the business logic have access to request headers (they can be accessed from `SessionState`), or should the request header information not leak to the business logic at all, but only be handled in the route handler layer?

Reply:
yes, we can handle in `get_session_state()`, `AppState` won't related to a session any more since its a global application state and header is session specific one. So lets not leak that information in `AppState`

Reply:
added `locale` in  `SessionState` 
 @SanchithHegde  @jarnura

Reply:
https://github.com/juspay/hyperswitch/pull/6933/files#diff-a6e49b399ffdee83dd322b3cb8bee170c65fb9c6e2e52c14222ec310f6927c54R115